src/main/java/org/jenkinsci/pipeline_steps_doc_generator/ProcessAsciiDoc.java

```diff
@@ -0,0 +1,93 @@
+package org.jenkinsci.pipeline_steps_doc_generator;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.commons.io.FilenameUtils;
+
+public class ProcessAsciiDoc {
+
+    public void generateHeader(StringBuilder toWrite, String param) {
+        toWrite.append("---\nlayout: pipelinesteps\ntitle: \"").append(param)
+                .append("\"\n---\n")
+                .append("== " + param + "\n")
+                .append("\n++++\n");
+    }
+
+    /*
+     * Takes in param name, and separates its documentation on to a new file.
+     */
+    public String separateParam(String param, String allAscii, File child)
+            throws IOException, RuntimeException {
+        BufferedReader br = new BufferedReader(new FileReader(child));
+        String line;
+        int counter = 0; // keeps a count of opening and closing li tags to mark the end of a section
+        boolean flag = false;
+        int lines = 0;
+        String url = param.toLowerCase().replaceAll("class", "").replaceAll("[^a-zA-Z0-9]", "");
+        Path adocPath = Path.of(allAscii + "/params/" + url + ".adoc"); // points to the params folder that will contain the separated files
+        StringBuilder newAdoc = new StringBuilder(); // new file that contains only the parameter details
+        StringBuilder duplicate = new StringBuilder(); // contains the content of the current file minus the parameter details
+        generateHeader(newAdoc, param);
+
+        while ((line = br.readLine()) != null) {
+            if (!flag && line.contains("<li><code>" + param + "</code><div>")) {
+                flag = true;
+                duplicate.append("<li><span><a href=\"/doc/pipeline/steps/params/" + url
+                        + "\" target=\"_blank\"><code>" + param + "</code></a></span></li>\n");
+            } else if (!flag) {
+                duplicate.append(line).append("\n");
+            }
+            if (flag) {
+                counter += (line.split("<li>", -1).length - line.split("</li>", -1).length);
```

Comment:
I'm not 100% sure what this is doing?  Is it looking for open and close tags?

Reply:
Yes, it is adding the difference between the number of opening and closing tags to see when a logical sequence of tags ends.